### DISCUSSION TITLE
PasswordStore State variable issue

### DISCUSSION THREAD

**Discussion by rocknwa** [Cyfrin/security-and-auditing-full-course-s23#171] [General]
In My audit report, I recommended using the `Immutable` keyword for the `s_owner` variable since it's stored in the contract's bytecode, I don't think it can be accessed off Chain because I used `cast` to test it, I couldn't access it. This is [My report](https://github.com/rocknwa/First-Security-Review/blob/main/report.pdf).

Please, I want to know if I'm right.

**Comment by danstam** [comment]
I'm guessing you meant using the `immutable` keyword for `s_password` instead of `s_owner`. However, this would not be the correct approach for the following reasons:

1- Declaring `s_password` as immutable would invalidate the contract's primary use case, which is to update a stored password. If the stored password is an immutable variable, it cannot be modified, rendering the contract's logic flawed.

2- While reading from bytecode would make accessing the password more difficult, it would not be impossible. All information stored on the blockchain is public, and it is never a good practice to store passwords on-chain in any form, even if declared as immutable.

**Reply by EngrPips** [reply]
Interesting, Does this answer clarify things @rocknwa?

**Comment by rocknwa** [comment]
Thanks a lot 😊

I really appreciate the clarification 🙏

**Reply by EngrPips** [reply]
Awesome. I'm glad you now understand it correctly.

**Reply by danstam** [reply]
You're welcome! I'm glad I could help.
